### DISCUSSION TITLE
Bulk Image generate with Fooocus

### DISCUSSION THREAD

**Discussion by cipher-x-sudo** [lllyasviel/Fooocus#3448] [Q&A] [answered]
Hey Everyone!

I need to know how can I generate bulk images. with fooocus. I have seen some issue in the community that were the same as I need, but I think they are outdated and I am not familiar with fooocus bulk image generation.

how can I generate bulk images like I save all the prompts to one of my file like txt file and that must be give to fooocus to generate images and it save the files automatically to my pc

**Comment by mashb1t** [comment, marked as answer]
You can use a wildcards file for that and process this one in order. You can find the checkbox in the Developer Debug mode and more information in the readme.

**Reply by cipher-x-sudo** [reply]
> You can use a wildcards file for that and process this one in order. You can find the checkbox in the Developer Debug mode and more information in the readme.

how do I implement them, tbh I am 0% familiar with what you are talking about

**Reply by mashb1t** [reply]
please check out https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#wildcards and ask your concrete answers afterwards. Thanks!

**Reply by cipher-x-sudo** [reply]
> please check out https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#wildcards and ask your concrete answers afterwards. Thanks!

The Wild cards are not my solution I guess. what i want is to batch process my prompts that I can store in files: TXT, CSV.

**Reply by cipher-x-sudo** [reply]
Wild Cards are like i guess to create dynamic responses. but I want is bulk creation via already made prompts

**Reply by mashb1t** [reply]
Use wildcards, process in order, then set image number to line count to process all prompts and generate one image for each line in the wildcards file.

**Reply by cipher-x-sudo** [reply]
TBH, I am really sorry i am disturb you again again. but i am totally noob with fooocus as well as wild card.

can you plz explain this process in steps. i have my prompts(Total : 312) in txt file. what to do and how. really sorry I am asking like that

**Reply by mashb1t** [reply]
Depending on your GPU i'd recommend you to work in batches of 32 images.
1. create a wildcard file (move your txt file to ghe wildcards folder) with all prompts, each prompt as single line
2. set image number to 32 (default maximum)
3. go to advanced > advanced > developer debug mode and check `Disable seed increment` and `Read wildcards in order`
4. edit your prompt and write `__filenameofyourwildcardsfile__` replacing the name with your filename
5. unselect any styles or select additional styles in advanced > styles
6. click generate

If you have a good GPU you may also increase the maximum image number in your config (see config example file for reference, copy the one line for max image number over and adjust the value), but keep in mind that all encoded image prompts will be kept in VRAM and might affect the performance of the other image generations if you use a too hogh batch count. I'd highly recommend to work in batches of 32 first before adjusting this.

Hope this helps!

**Reply by cipher-x-sudo** [reply]
> Depending on your GPU i'd recommend you to work in batches of 32 images.
> 
> 1. create a wildcard file (move your txt file to ghe wildcards folder) with all prompts, each prompt as single line
> 2. set image number to 32 (default maximum)
> 3. go to advanced > advanced > developer debug mode and check `Disable seed increment` and `Read wildcards in order`
> 4. edit your prompt and write `__filenameofyourwildcardsfile__` replacing the name with your filename
> 5. unselect any styles or select additional styles in advanced > styles
> 6. click generate
> 
> If you have a good GPU you may also increase the maximum image number in your config (see config example file for reference, copy the one line for max image number over and adjust the value), but keep in mind that all encoded image prompts will be kept in VRAM and might affect the performance of the other image generations if you use a too hogh batch count. I'd highly recommend to work in batches of 32 first before adjusting this.
> 
> Hope this helps!

I have a 3080-OC and still trying this

**Reply by cipher-x-sudo** [reply]
i have 312 prompts in a text file how can I do it in batch as you said 32 is max

**Reply by cipher-x-sudo** [reply]
> If you have a good GPU you may also increase the maximum image number in your config (see config example file for reference, copy the one line for max image number over and adjust the value), but keep in mind that all encoded image prompts will be kept in VRAM and might affect the performance of the other image generations if you use a too hogh batch count. I'd highly recommend to work in batches of 32 first before adjusting this.

can you please explain more\

**Reply by netenet** [reply]
> Depending on your GPU i'd recommend you to work in batches of 32 images.
> 
> 1. create a wildcard file (move your txt file to ghe wildcards folder) with all prompts, each prompt as single line
> 2. set image number to 32 (default maximum)
> 3. go to advanced > advanced > developer debug mode and check `Disable seed increment` and `Read wildcards in order`
> 4. edit your prompt and write `__filenameofyourwildcardsfile__` replacing the name with your filename
> 5. unselect any styles or select additional styles in advanced > styles
> 6. click generate
> 
> If you have a good GPU you may also increase the maximum image number in your config (see config example file for reference, copy the one line for max image number over and adjust the value), but keep in mind that all encoded image prompts will be kept in VRAM and might affect the performance of the other image generations if you use a too hogh batch count. I'd highly recommend to work in batches of 32 first before adjusting this.
> 
> Hope this helps!

It is not working, I followed all the steps you mentioned but no, please explain

**Reply by netenet** [reply]
Waiting for your reply
